hooks/pre_build

```diff
@@ -0,0 +1,27 @@
+#!/bin/bash
+#
+# This script is intended for Docker Hub Automated Builds since we can't
+# build on TaskCluster because of too old Docker version (see issue #3057)
+#
+# Docker Hub expects an existing ready-to-use Dockerfile, and this hook allows
+# to generate one basing on the existing |make| template, but we need to change
+# in-place, hence why the |cp| statement
+#
+# Docker Hub Automated Builds needs to be configured as:
+#
+# Source Type | Source                | Docker Tag | Dockerfile Location   | Build Context
+# ----------------------------------------------------------------------------------------
+# Tag         | /^v([0-9.\-a-zA-Z]+)/ | v{\1}      | Dockerfile.train.tmpl | /
+# Tag         | /^v([0-9.\-a-zA-Z]+)/ | v{\1}      | Dockerfile..tmpl | /
+
+set -e # Exit immediately if a command exits with a non-zero status.
+set -u # Treat unset variables as an error.
+
+DOCKERFILE_TARGET=$(echo ${DOCKERFILE_PATH} | sed -e 's/\.tmpl//g')
```

Comment:
How is `DOCKERFILE_PATH` populated? Maybe add a brief comment?

Reply:
Done, explanation on env vars + link to the doc